go/sql/types.go

```diff
@@ -244,6 +245,19 @@ func (this *ColumnList) IsSubsetOf(other *ColumnList) bool {
 	return true
 }
 
+func (this *ColumnList) FilterBy(f func(Column) bool) *ColumnList {
```

Comment:
Since we're creating a new instance of `ColumnList`, what if we turn this method into factory function? Something like:

```go 
 NewFilteredColumnList(ordinals ColumnsMap, f func(Column) bool) *ColumnList {
 ...
}
```